HIP/hip-1082.md

```diff
@@ -0,0 +1,225 @@
+    alias (long zero) and an EVM address alias (standard ECDSA derived EVM address).
+- When a user interacts with the EVM the network will utilize the ECDSA derived EVM address alias on an account if
+    present. If not it will utilize the long zero.
+- Accounts with an ECDSA derived EVM address are fully compatible with expectations of contracts that utilize the
+    ecrecover precompile.
+- Accounts without an ECDSA derived EVM address alias will not match the expectation of the ecrecover precompile
+    calculations. This has effects on certain DApp behaviour.
+- Smart contracts can and do cache the evm address of an account and often use it to assign attributes such as access
+    and ownership. As a result it was important to maintain a static alias to avoid unintended differences in
+    expectation
+- Account key rotation has no impact on the evm address utilized when an account interacts with the EVM
+
+<aside>
+🚨 **Open Task:** : Add account, address key relationship diagram for clarity
+</aside>
+
+## User stories
+
+1. As an existing account with an ED key but no EVM address alias, I would like to set an ECDSA derived EVM address
+    alias on my account that will identify my account on the EVM.
+2. As an existing account with an EC key but no EVM address alias, I would like to set an ECDSA derived EVM address
+    alias on my account that will identify my account on the EVM.
+3. As an existing account with a complex key but no EVM address alias, I would like to set an ECDSA derived EVM address
+    alias on my account that will identify my account on the EVM.
+5. As an existing account with a newly set ECDSA derived EVM address, I want my account to conform with `ecrecover`
+    expectations as long as I sign the transaction with the ECDSA key that corresponds to the EVM address alias.
+    This key may or may not be the key currently on my account.
+5. As an account with an EVM address alias set after creation, I would like to set an evm address override to my
+    previous long zero address for a `ContractCreate` transaction.
+6. As an account with an EVM address alias set after creation, I would like to set an evm address override to my
+    previous long zero address for a `ContractCall` transaction.
+  
+## Specification
+
+The main change to the network is simple - `CryptoUpdate` transactions will support setting a new ECDSA derived
+`evm_address` if the current value is unset.
+
+### Protobuf
+```protobuf
+message CryptoUpdateTransactionBody {
+    ...
+
+    /**
+     * EOA 20-byte address to create that is derived from the keccak-256 hash of a ECDSA_SECP256K1 primitive key.
+     */
+    bytes evm_address = 19;
+}
+```
+
+Additionally, `ContractCreate` and `ContractCall` EVM transactions will need to support the ability to specify the
+Hedera Num Alias address to use on the transaction. This is because on the HAPI side all aliases are applicable and
+thus the flexibility of the API is important to maintain.
```

Comment:
I still don't understand why the HIP allows them to set an ECDSA derived EVM address on their existing account but then it doesn't allow them to use this new value on contract calls and instead requires the long zero form.

And even if Hedera Num Alias is required, why would they specify it as long zero bytes instead of just using `AccountID` in shard.realm.num?

Reply:
If a valid alias is set where one did not exist the new value is indeed used during Contract calls.
The override option is provided to set the older long-zero address in the case that a user would like to be perceived by the EVM as such this to unblock themselves.
This is why the address format is used and not the AccountID format.

For example, say an account with no alias has interacted with a contract that cached its long zero address. If they set the alias with this HIP and again interacted with the contract they would likely see the new address cached.
If for some reason that account wanted to migrate in contract details from the original long zero address to the new ECDSA derived address they could do so by utilizing this override in addition to the contract logic.